### PR TITLE
Personal/cramavarapu/sas token copier propagation

### PR DESCRIPTION
**Pull request by cyrus-ramavarapu-ms**
Copytool copier library should be able to access refreshed SAS tokens

This commit caches 403 Forbidden errors at the copier layer that can
result from SAS token expiration during a download or upload. When the
copier encounters this error it (or the faulting go routine) call a
call back into the mover to force a refresh of the SAS token and
recreation of both the container client and blockblob client.

As a minor change, debug print messages were added to the copier along error paths in order to simplify investigation.

### CODE REVIEW

File: cmd/lhsm-plugin-az/mover.go [hunk 1]
```diff
@@ -336,12 +363,13 @@ func (m *Mover) Restore(ctx context.Context, action dmplugin.Action) error {
 	}
 
 	contentLen, err := core.Restore(ctx, m.copier, core.RestoreOptions{
-		ContainerURL:    cURL,
-		BlobName:        fileKey,
-		DestinationPath: action.WritePath(),
-		BlockSize:       m.config.UploadPartSize,
-		ExportPrefix:    m.config.ExportPrefix,
-		HTTPClient:      m.httpClient,
+		ContainerURL:			cURL,
```
Comment (tong-xiao-ms) [inline cmd/lhsm-plugin-az/mover.go:366]: Format here is not at the same level.

Comment (cyrus-ramavarapu-ms) [reply cmd/lhsm-plugin-az/mover.go]: Fixed.

File: cmd/lhsm-plugin-az-core/archive.go [hunk 2]
```diff
@@ -107,9 +108,20 @@ func Archive(ctx context.Context, copier copier.Copier, o ArchiveOptions) (int64
 
 			blob := o.ContainerURL.NewBlockBlobClient(blobpath)
 			_, err = blob.UploadBuffer(ctx, nil, options)
+			if util.ShouldRefreshCreds(err) {
+				o.ContainerURL, blob, err = o.GetNewStorageClients(blobpath)
+				if err != nil {
+					util.Log(pipeline.LogError, fmt.Sprintf("Failed to get new block blob client for %s: %v", blobpath, err))
+					return
+				}
+				blob.UploadBuffer(ctx, nil, options)
```
Comment (ellis-wilson-ms) [inline cmd/lhsm-plugin-az-core/archive.go:117]: Any risk here that not storing err like we do on 110 can result in the func exiting as if it succeeded when it actually failed?  I'll be honest, Go here has me completely lost on how err gets persisted whatsoever in this goroutine.

File: cmd/lhsm-plugin-az-core/archive.go [hunk 3]
```diff
@@ -15,21 +15,22 @@ import (
...
-	MountRoot    string
-	FID          string
-	BlobName     string
-	SourcePath   string
-	BlockSize    int64
-	ExportPrefix string
-	HTTPClient   *http.Client
-	OpStartTime  time.Time
+	ContainerURL          *container.Client
+	ResourceSAS           string
```
[the remark this answers is not in the document]
Comment (cyrus-ramavarapu-ms) [reply cmd/lhsm-plugin-az-core/archive.go:24]: I don't know what that is and it didn't come in with my changes. It exists in master as well.

[the remark this answers is not in the document]
Comment (ellis-wilson-ms) [reply cmd/lhsm-plugin-az-core/archive.go:24]: Resolving this.  We're in comfort care mode with this tool, so let's not pull on any stitches too hard.

**Pull request merged**